twine/settings.py

```diff
@@ -0,0 +1,280 @@
+        )
+        parser.add_argument(
+            "--verbose",
+            action="store_true",
+            help="Show verbose output."
+        )
+
+    @classmethod
+    def from_argparse(cls, args):
+        """Generate the Settings from parsed arguments."""
+        settings = vars(args)
+        settings['repository_name'] = settings.pop('repository')
+        settings['cacert'] = settings.pop('cert')
+        return cls(**settings)
+
+    def _handle_package_signing(self, sign, sign_with, identity):
+        if not sign and identity:
+            raise exceptions.InvalidSigningConfiguration(
+                "sign must be given along with identity"
+            )
+        self.sign = sign
+        self.sign_with = sign_with
+        self.identity = identity
+
+    def _handle_repository_options(self, repository_name, repository_url):
+        self.repository_config = utils.get_repository_from_config(
+            self.config_file,
+            repository_name,
+            repository_url,
+        )
+        self.repository_config['repository'] = utils.normalize_repository_url(
+            self.repository_config['repository'],
+        )
+
+    def _handle_authentication(self, username, password):
+        self.username = utils.get_username(username, self.repository_config)
+        self.password = utils.get_password(
+            self.repository_config['repository'],
+            self.username,
+            password,
+            self.repository_config,
+        )
+
+    def _handle_certificates(self, cacert, client_cert):
+        self.cacert = utils.get_cacert(cacert, self.repository_config)
+        self.client_cert = utils.get_clientcert(
+            client_cert,
+            self.repository_config,
+        )
+
+    def check_repository_url(self):
+        """Verify we are not using legacy PyPI.
+
+        :raises:
+            :class:`~twine.exceptions.UploadToDeprecatedPyPIDetected`
+        """
+        repository_url = self.repository_config['repository']
+
+        if repository_url.startswith((repository.LEGACY_PYPI,
+                                      repository.LEGACY_TEST_PYPI)):
+            raise exceptions.UploadToDeprecatedPyPIDetected(
+                "You're trying to upload to the legacy PyPI site '{0}'. "
+                "Uploading to those sites is deprecated. \n "
+                "The new sites are pypi.org and test.pypi.org. Try using "
+                "{1} (or {2}) to upload your packages instead. "
+                "These are the default URLs for Twine now. \n More at "
+                "https://packaging.python.org/guides/migrating-to-pypi-org/ "
+                ".".format(
+                    repository_url,
+                    utils.DEFAULT_REPOSITORY,
+                    utils.TEST_REPOSITORY
+                    )
+                )
+
+    def create_repository(self):
+        """Create a new repository for uploading."""
+        repo = repository.Repository(
```

Comment:
This logic was already being repeated, so I pulled it out to here. The same goes for the logic in check_repository_url.